dv/uvm/core_ibex/scripts/compile-tb.py

```diff
@@ -35,6 +37,32 @@ def main() -> int:
 
     os.makedirs(args.output, exist_ok=True)
 
+    # Setup environment variables for the tb to link against the compiled ISS
+    if args.en_cosim:
+        _spike_iss_pc = 'riscv-riscv riscv-disasm riscv-fdt'
+
+        if not subprocess.run(shlex.split(f'pkg-config --exists {_spike_iss_pc}'),
+                              check=True):
+            raise EnvironmentError(f'Failed to find "{_spike_iss_pc}" pkg-config\
+            packages. Did you set the PKG_CONFIG_PATH correctly?')
+
+        def _get_iss_pkgconfig_flags(cmd, iss_pc):
+            _flag = subprocess.check_output(
+                args=shlex.split(f'pkg-config {cmd} {iss_pc}'),
+                universal_newlines=True)
+            if args.simulator == 'xlm':
+                return _flag.replace('-Wl,', '')  # Don't ask
+            else:
+                return _flag
+
+        _iss_pkgconfig_envvars_dict = {
+            'ISS_CFLAGS'  : '--cflags',
+            'ISS_LDFLAGS' : '--libs-only-other',
+            'ISS_LIBS'    : '--libs-only-l --libs-only-L',
+        }
+        for k, v in _iss_pkgconfig_envvars_dict.items():
+            os.environ[k] = _get_iss_pkgconfig_flags(v, _spike_iss_pc)
```

Comment:
The usual approach here is to take a copy of the environment and modify that, then pass it to subprocess (I think `run_one` has an `env` argument that you can use)